plugins/modules/proxmox_backup.py

```diff
@@ -351,23 +325,11 @@ def check_vmid_backup_permission(self, permissions, vmids, pool):
             sufficient_permissions = True
         # Finally, when no check succeeded, fail
         if not sufficient_permissions:
-            self.module.fail_json(
-                changed=False,
-                msg="Insufficient permissions: You do not have the VM.Backup permission")
+            self.module.fail_json(changed=False, msg="Insufficient permissions: You do not have the VM.Backup permission")
 
     def check_general_backup_permission(self, permissions, pool):
-        if "/" in permissions.keys() and permissions["/"].get(
-                "VM.Backup", 0) == 1:
-            pass
-        elif "/vms" in permissions.keys() and permissions["/vms"].get("VM.Backup", 0) == 1:
-            pass
-        elif pool and "/pool/" + pool in permissions.keys() and permissions["/pool/" + pool].get(
-                "VM.Backup", 0) == 1:
-            pass
-        else:
-            self.module.fail_json(
-                changed=False,
-                msg="Insufficient permissions: You dont have the VM.Backup permission")
+        if not has_permission(permissions, "VM.Backup", search_scopes=["/", "/vms"] + (["/vms" + pool] if pool else [])):
```

Comment:
search_scopes should be ` ["/", "/vms"] + (["/pool" + pool] if pool else [])`

Reply:
```suggestion
        if not has_permission(permissions, "VM.Backup", search_scopes=["/", "/vms"] + (["/pool" + pool] if pool else [])):
```